stackit/internal/services/cdn/customdomain/resource.go

```diff
@@ -260,30 +314,142 @@ func (r *customDomainResource) ImportState(ctx context.Context, req resource.Imp
 	tflog.Info(ctx, "CDN custom domain state imported")
 }
 
-func mapCustomDomainFields(customDomain *cdn.CustomDomain, model *CustomDomainModel) error {
-	if customDomain == nil {
+func normalizeCertificate(certInput cdn.GetCustomDomainResponseGetCertificateAttributeType) (Certificate, error) {
+	var customCert *cdn.GetCustomDomainCustomCertificate
+	var managedCert *cdn.GetCustomDomainManagedCertificate
+
+	if certInput == nil {
+		return Certificate{}, errors.New("input of type GetCustomDomainResponseCertificate is nil")
+	}
+	customCert = certInput.GetCustomDomainCustomCertificate
+	managedCert = certInput.GetCustomDomainManagedCertificate
+
+	// Now we process the extracted certificates
+	if customCert != nil && customCert.Type != nil && customCert.Version != nil {
+		return Certificate{
+			Type:    *customCert.Type,
+			Version: customCert.Version, // Converts from *int64 to int
+		}, nil
+	}
+
+	if managedCert != nil && managedCert.Type != nil {
+		// The version will be the zero value for int (0), as requested
+		return Certificate{
+			Type: *managedCert.Type,
+		}, nil
+	}
+
+	return Certificate{}, errors.New("certificate structure is empty, neither custom nor managed is set")
+}
+
+// buildCertificatePayload constructs the certificate part of the payload for the API request.
+// It defaults to a managed certificate if the certificate block is omitted, otherwise it creates a custom certificate.
+func buildCertificatePayload(ctx context.Context, model *CustomDomainModel) (*cdn.PutCustomDomainPayloadCertificate, diag.Diagnostics) {
```

Comment:
Replace the return type with error.
```suggestion
func buildCertificatePayload(ctx context.Context, model *CustomDomainModel) (*cdn.PutCustomDomainPayloadCertificate, error) {
```

If you have within the buildCertificatePayload an error from diags, return it like this:
```go
	if diags.HasError() {
		return fmt.Errorf("invalid certificate or private key: %w", core.DiagsToError(diags))
	}
```

And in the place where you call the function buildCertificatePayload(), e.g. in `Create()` you can handle the error like this:
```go
_, err := buildCertificatePayload(ctx, model)
if err != nil {
		core.LogAndAddError(ctx, &resp.Diagnostics, "Error creating CDN custom domain", fmt.Sprintf("Creating API payload: %v", err))
		return
}
```